lib/async.js

```diff
@@ -34,7 +34,7 @@ module.exports = function resolve(x, options, callback) {
 
     if (/^(?:\.\.?(?:\/|$)|\/|([A-Za-z]:)?[/\\])/.test(x)) {
         var res = path.resolve(y, x);
-        if (x === '..') res += '/';
+        if (x === '..' || x[x.length - 1] === '/') res += '/';
```

Comment:
indexing into strings doesn't work in IE 6-8 (it crashes pretty hard) - i'd prefer to use `.slice(-1)` or `.charAt(x.length - 1)` here.